src/mongo_odm/model.hpp

```diff
@@ -35,6 +35,25 @@ struct FirstTypeIsTheSame : public std::false_type {};
 template <typename T, typename T2, typename... Ts>
 struct FirstTypeIsTheSame<T, T2, Ts...> : public std::is_same<T, std::decay_t<T2>> {};
 
+/**
+ * Type trait that checks whether or not a type is a container that contains a particular type.
+ *
+ * @tparam container_type The container being checked for containing a particular type.
+ * @tparam T The type that the container must hold for ContainerOfT::value to be true.
+ */
+template <typename container_type, typename T>
+struct ContainerOfT : public std::is_same<typename container_type::value_type, T> {};
```

Comment:
can all of these type traits be moved into `utils.hpp` with the other type traits?

Also, for consistency, rename to snake case (and maybe `is_container_of`), and then offer `is_container_of_v`